src/main/java/com/google/sps/servlets/ContextServlet.java

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.youtube.YouTube;
+import com.google.api.services.youtube.model.VideoListResponse;
+import com.google.gson.Gson;
+import com.google.sps.servlets.utils.VideoInformation;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that fetches from Youtube Server. */
+@WebServlet("/VideoContext")
+public class ContextServlet extends HttpServlet {
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
```

Comment:
Ideally we should figure out a way to share this information between the `YoutubeServlet` and the `ContextServlet`.

I'm okay with doing this in a follow-up PR since the `YoutubeServlet` is currently in flux, but generally you should try not to have the same logic/config live in two different places since it makes maintenance + future enhancements more difficult.

This kind of approach can be referred to as following the "DRY" (Don't repeat yourself) principle which is something that you can read about if you're interested. (https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)

BTW after typing this out I noticed your TODO...

Reply:
LOL. I agree with Rick. Maybe it is worth it to wait on Uri submitting his side of the code(almost done) and we can think about merging of the two. 
Xin mentioned earlier that she wanted the servlets to run separately so that they can get the context but it shouldn't matter if you have async calls.

Reply:
haha no problem! 
I will wait for Uri to submit his code to encapsulate getService(), JSON_Factory etc in a separate file. (Is that what we want haha)